v2/internal/controllers/crd_networking_applicationgateway_test.go

```diff
@@ -0,0 +1,389 @@
+			RequestRoutingRules: []network.ApplicationGatewayRequestRoutingRule{
+				{
+					Name: to.Ptr[string]("app-gtw-routing-rule-1"),
+					BackendAddressPool: &network.ApplicationGatewaySubResource{
+						Reference: &genruntime.ResourceReference{
+							ARMID: appGtwBackendPoolsID,
+						},
+					},
+					BackendHttpSettings: &network.ApplicationGatewaySubResource{
+						Reference: &genruntime.ResourceReference{
+							ARMID: appGtwBackendHttpSettingsID,
+						},
+					},
+					HttpListener: &network.ApplicationGatewaySubResource{
+						Reference: &genruntime.ResourceReference{
+							ARMID: appGtwFHttpListnerID,
+						},
+					},
+					RuleType: to.Ptr[network.ApplicationGatewayRequestRoutingRulePropertiesFormat_RuleType](network.ApplicationGatewayRequestRoutingRulePropertiesFormat_RuleType(network.ApplicationGatewayRequestRoutingRulePropertiesFormat_RuleType_Basic)),
+					Priority: to.Ptr[int](100),
+				},
+			},
+		},
+	}
+	tc.CreateResourceAndWait(applicationGateway)
+	tc.Expect(applicationGateway.Status.Id).ToNot(BeNil())
+	tc.Expect(applicationGateway.Status.Sku.Tier).To(Equal(network.ApplicationGatewaySku_Tier_STATUS_WAF_V2))
+	armId := *applicationGateway.Status.Id
+	tc.DeleteResourceAndWait(applicationGateway)
+	tc.DeleteResourceAndWait(publicIPAddress)
+	tc.DeleteResourceAndWait(subnet)
+	tc.DeleteResourceAndWait(vnet)
+	// Ensure that the resource was really deleted in Azure
+	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(network.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func defineApplicationGatewayIPConfiguration(subnet *genruntime.ResourceReference) []network.ApplicationGatewayIPConfiguration_ApplicationGateway_SubResourceEmbedded {
+	subResName := "app-gw-ip-config-1"
+	appGtwFeIpConfig := []network.ApplicationGatewayIPConfiguration_ApplicationGateway_SubResourceEmbedded{
+		{
+			Name: to.Ptr[string](subResName),
+			Subnet: &network.ApplicationGatewaySubResource{
+				Reference: subnet,
+			},
+		},
+	}
+
+	return appGtwFeIpConfig
+}
+
+func defineApplicationGatewayWebApplicationFirewallConfiguration(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, appGatewayName string) (*network.ApplicationGatewayWebApplicationFirewallConfiguration, string) {
+	subResname := "app-gw-waf-config-1"
+	subRes := to.Ptr[network.ApplicationGatewayWebApplicationFirewallConfiguration](network.ApplicationGatewayWebApplicationFirewallConfiguration{
+		Enabled:        to.Ptr[bool](true),
+		FirewallMode:   to.Ptr[network.ApplicationGatewayWebApplicationFirewallConfiguration_FirewallMode](network.ApplicationGatewayWebApplicationFirewallConfiguration_FirewallMode(network.ApplicationGatewayWebApplicationFirewallConfiguration_FirewallMode_Detection)),
+		RuleSetType:    to.Ptr[string]("OWASP"),
+		RuleSetVersion: to.Ptr[string]("3.2"),
+	})
+	armid_params := map[string]string{
```

Comment:
```suggestion
	armIDParams := map[string]string{
```

Reply:
same below for all the usages like this. Though I think it'd be cleaner to be a method with 4 args (see my above comment) instead.